src/analysis/normed_space/spectrum.lean

```diff
@@ -0,0 +1,117 @@
+/-
+Copyright (c) 2021 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import algebra.algebra.spectrum
+import analysis.calculus.deriv
+/-!
+# The spectrum of elements in a complete normed algebra
+
+This file contains the basic theory for the resolvent and spectrum of a Banach algebra.
+
+## Main definitions
+
+* `spectral_radius : ℝ≥0∞`: supremum of `∥k∥₊` for all `k ∈ spectrum 𝕜 a`
+
+## Main statements
+
+* `is_open_resolvent_set`: the resolvent set is open.
+* `is_closed`: the spectrum is closed.
+* `subset_closed_ball_norm`: the spectrum is a subset of closed disk of radius equal to the norm.
+* `is_compact`: the spectrum is compact.
+* `spectral_radius_le_nnnorm`: the spectral radius is bounded above by the norm.
+* `has_deriv_at_resolvent`: the resolvent function is differentiable on the resolvent set.
+
+
+## TODO
+
+* after we have Liouville's theorem, prove that the spectrum is nonempty when the
+  scalar field is ℂ.
+* compute all derivatives of `resolvent a`.
+
+-/
+
+open_locale ennreal
+
+/-- The *spectral radius* is the supremum of the `nnnorm` (`∥⬝∥₊`) of elements in the spectrum,
+    coerced into an element of `ℝ≥0∞` so that it lives in a `complete_lattice`. Note that it
+    is possible for `spectrum 𝕜 a = ∅`. In this case, `spectral_radius a = 0`-/
+noncomputable def spectral_radius (𝕜 : Type*) {A : Type*} [normed_field 𝕜] [ring A]
+  [algebra 𝕜 A] (a : A) : ℝ≥0∞ :=
```

Comment:
Wouldn't it make more sense for this to be a `nnreal` rather than a `ennreal`, since it's never infinite? (i.e. I don't see why it would be important for this to live in a complete lattice.)

Reply:
A few reasons:

1. It gives you access to all the theorems about `supr` on complete lattices.
2. you can always reduce to `nnreal` if need be with `mod_cast` tactics since the definition casts it into an `ennreal` in first place (see, e.g., the proof of `spectral_radius_le_nnnorm`).
3. perhaps most importantly: unbounded operators can have unbounded spectrum, and at least that tau-measurable operators (in general unbounded) affiliated with a semifinite von Neumann algebra with faithful normal semifinite trace tau form a *-algebra. So really, I should have been allowing infinity in the first place anyway.

Reply:
To really drive home the last point in a way which is probably more accessible: consider L^0([0,1]), the collection of measurable functions on the unit interval which are finite almost everywhere, with the usual quotient by equal almost everywhere equivalence. This is an algebra, and the spectrum of an element is precisely its essential range. So, if the element isn't L^infty, then its spectrum is unbounded.

Reply:
I find this last example really convincing -- and like Frederic I had been initially dubious when the rationale was expressed as "making it live in a complete lattice".  Let me suggest-change a docstring which emphasizes these points.